### PR TITLE
planner: PhysicalPlan memory trace 4

### PR DESCRIPTION
**Pull request by fzzf678**
### What problem does this PR solve?

Issue Number: ref #37632 
Problem Summary:

### What is changed and how it works?
Add MemoryUsage() for
* PhysicalTableReader
* PhysicalIndexReader
* PhysicalIndexMergeReader
* PhysicalIndexLookUpReader

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: executor/builder.go [hunk 1]
```diff
@@ -3922,6 +3922,15 @@ type mockPhysicalIndexReader struct {
 	e Executor
 }
 
+// MemoryUsage return the memory usage of mockPhysicalIndexReader
+func (p *mockPhysicalIndexReader) MemoryUsage() (sum int64) {
```
Comment (qw4990) [inline executor/builder.go:3926]: It seems like this Plan is only used for testing, so we can just return 0 here.

File: planner/core/physical_plans.go [hunk 2]
```diff
@@ -186,6 +186,20 @@ func (p *PhysicalTableReader) GetAvgRowSize() float64 {
...
 }
 
+// MemoryUsage return the memory usage of PhysicalTableReader
+func (p *PhysicalTableReader) MemoryUsage() (sum int64) {
+	if p == nil {
+		return
+	}
+
+	sum = p.physicalSchemaProducer.MemoryUsage() + size.SizeOfUint8*2 + size.SizeOfBool + p.PartitionInfo.MemoryUsage()
+	// todo: memtrace: 	p.TablePlans p.PhysicalPlan
```
Comment (qw4990) [inline planner/core/physical_plans.go:196]: Why not add the function `MemoryUsage() (sum int64)` to the `PhysicalPlan` interface? 
Then we can accumulate their memory usage like `sum += p.tablePlan.MemoryUsage()`.

**Comment by fzzf678** [comment]
@qw4990 PTAL again

**Review by qw4990 — APPROVED** [review]

**Review by Reminiscent — APPROVED** [review]

**Pull request merged**